README.md

```diff
@@ -1,5 +1,86 @@
-UoN.AspNetCore.VersionMiddleware
-===============
+# UoN.AspNetCore.VersionMiddleware
+
 [![License](https://img.shields.io/badge/licence-MIT-blue.svg)](https://opensource.org/licenses/MIT)
 
-Version middleware for ASP.NET Core.
+## What is it?
+
+This is middleware for ASP.Net Core designed to report on the assembly informational version of your web project's assembly.
+
+We use it at UoN so that we can check the version of a web app wherever it's deployed, without having to display it publicly to users who don't care. This is useful for ensuring testers know which builds they're working with and therefore what fixes to test.
+
+## What are its features?
+
+It provides three `IApplicationBuilder` Extension methods for you to use in `Startup.Configure()`:
+
+- `app.UseVersion()`
+  - adds a `/version` route to the ASP.Net Core pipeline.
+  - this returns as plaintext the informational version of the entry assembly (i.e. containing this `Startup` class.)
+- `app.MapVersion(path)`
+  - adds a route with the specified path string to the ASP.Net Core pipeline
+  - behaves otherwise the same as above
+- `app.MapVersion(path, assembly)`
+  - behaves the same as above, but gives the version of the specified assembly instead of the entry assembly.
+
+## Dependencies
+
+The library targets `netstandard2.0` and depends upon ASP.Net Core 2.0.
+
+If you can use ASP.Net Core 2, you can use this library.
+
+## Usage
+
+Acquire the library via one of the methods below, and add one of the above extension methods to your ASP.Net Core pipeline.
+
+### NuGet
+
+This library is hosted on nuget.org <TODO: add link>
```

Comment:
Change this to `It will be` and then add an issue for adding it to nuget.org and remove the todo 🗡

Reply:
updated and created #8